task8-rashid-ab/src/Producer.cpp

```diff
@@ -0,0 +1,35 @@
+#include "Producer.h"
+#include <iostream>
+
+std::mutex mtx;
+std::condition_variable cv;
+bool finished = false;
+
+/**
+ * Constructs a Producer object with the given thread ID and value.
+ *
+ * @param id   The ID of the thread associated with the producer.
+ * @param val  The initial value for the producer.
+ */
+Producer::Producer(std::thread::id id, double val) : threadId(id), value(val) {}
+
+/**
+ * Runs the producer, generating values and printing them to the console.
+ * Once the value becomes zero, it signals completion and waits for other producers to finish.
+ */
+void Producer::run() {
+    std::unique_lock<std::mutex> lock(mtx);  // Acquire the unique lock on the mutex
+    while (value != 0) {
+        std::cout << std::hash<std::thread::id>{}(threadId) << " sent: " << value << std::endl;
```

Comment:
The producer needs to actually send data to the consumer. The exercise demand was that the consumer will be the one to print the data